Fix: The latest GA should be updated in the stable tag [ED-10526]

maork-elementor opened the pull request.

maork-elementor committed:
test

maork-elementor committed:
fix syntax

maork-elementor committed:
debug

maork-elementor committed:
git fetch --tags

maork-elementor committed:
fix

maork-elementor committed:
fix

maork-elementor committed:
fix

maork-elementor committed:
syntax

maork-elementor committed:
fix if

maork-elementor committed:
fix

maork-elementor committed:
update file path

maork-elementor committed:
Merge branch 'main' into ED-10526

maork-elementor committed:
not cloud

yotamselementor commented on .github/workflows/update-readme-txt/action.yml:
```diff
@@ -0,0 +1,36 @@
+        run: |
+              # Get the latest ga tag e.g. 3.13.2
+              if [[ "${{ inputs.CHANNEL }}" == "cloud" || "${{ inputs.CHANNEL }}" == "beta" || "${{ inputs.CHANNEL }}" == "dev" ]]; then
+                latest_tag=$(git ls-remote --tags | grep -v "\-rc" | grep -v "\-cloud" | grep -v "\-dev" | grep -v "\-beta" | awk '{print $2}' | grep -E 'v[0-9]+\.[0-9]+\.[0-9]+$' | sort -V | tail -n 1 | sed 's/^refs\/tags\/v//')
```
Maybe add a comment explaining this as well

maork-elementor replied:
I added support for with "v" and without "v". (With the current PR code changes)

`git ls-remote --tags | grep -v "\-rc" | grep -v "\-cloud" | grep -v "\-dev" | grep -v "\-beta" | awk '{print $2}' | grep -E 'v[0-9]+\.[0-9]+\.[0-9]+$' | sort -V | tail -n 1 | sed 's/^refs\/tags\/v//'`

I tried the Commit suggestion, didn't get any result (directly on my Terminal).
You can run this line on your local and point it to Elementor remote for testing : 
(Add remote name after the `tags` option)

`git ls-remote --tags elementor | grep -v "\-rc" | grep -v "\-cloud" | grep -v "\-dev" | grep -v "\-beta" | awk '{print $2}' | grep -E 'v[0-9]+\.[0-9]+\.[0-9]+$' | sort -V | tail -n 1 | sed 's/^refs\/tags\/v//'`

yotamselementor replied:
git ls-remote --tags | grep -v "\-rc" | grep -v "\-cloud" | grep -v "\-dev" | grep -v "\-beta" | awk '{print $2}' | grep -E 'v[0-9]+\.[0-9]+\.[0-9]+$' | sort -V | tail -n 1 | sed 's/^refs\/tags\/v//'

<img width="929" alt="image" src="https://user-images.githubusercontent.com/83912471/234814313-5cf86d92-2363-408e-a348-db615410eaeb.png">

maork-elementor replied:
You can give this a try : 

`git ls-remote --tags | grep -v "\-rc" | grep -v "\-cloud" | grep -v "\-dev" | grep -v "\-beta" | awk '{print $2}' | grep -E '[0-9]+\.[0-9]+\.[0-9]+$' | sed 's/^refs\/tags\/v//' | sed 's/^refs\/tags\///' | sort -V | tail -n 1`

yotamselementor commented on .github/workflows/update-readme-txt/action.yml:
```diff
@@ -0,0 +1,36 @@
+
+              sed -i -E "s/Stable tag: .*/Stable tag: ${latest_tag}/g" $GITHUB_WORKSPACE/elementor/readme.txt
+
+              echo "readme.txt updated with Stable tag: ${latest_tag}"
```
Print the readme.txt file for debug

maork-elementor replied:
Done

yotamselementor commented on .github/workflows/update-readme-txt/action.yml:
```diff
@@ -0,0 +1,36 @@
+              sed -i -E "s/Stable tag: .*/Stable tag: ${latest_tag}/g" $GITHUB_WORKSPACE/elementor/readme.txt
+
+              echo "readme.txt updated with Stable tag: ${latest_tag}"
+
+
+              
+              
+
```
```suggestion

```

maork-elementor replied:
Done

yotamselementor commented on .github/workflows/get-previous-release/action.yml:
Why did you change this file?

maork-elementor replied:
To improve the logic & support with "v" and without.

maork-elementor replied:
Separated to Separated PR

maork-elementor commented on .github/workflows/update-readme-txt/action.yml:
```diff
@@ -0,0 +1,36 @@
+  PACKAGE_VERSION:
+    required: true
+    description: 'The package version to get the version from.'
+
+
```
Done

yotamselementor replied:
No, its not

maork-elementor replied:
Sorry, now the change is really pushed.

maork-elementor commented on .github/workflows/update-readme-txt/action.yml:
```diff
@@ -0,0 +1,36 @@
+              # Get the latest ga tag e.g. 3.13.2
+              if [[ "${{ inputs.CHANNEL }}" == "cloud" || "${{ inputs.CHANNEL }}" == "beta" || "${{ inputs.CHANNEL }}" == "dev" ]]; then
+                latest_tag=$(git ls-remote --tags | grep -v "\-rc" | grep -v "\-cloud" | grep -v "\-dev" | grep -v "\-beta" | awk '{print $2}' | grep -E 'v[0-9]+\.[0-9]+\.[0-9]+$' | sort -V | tail -n 1 | sed 's/^refs\/tags\/v//')
+              fi
+
+              # Get the latest ga tag e.g. 3.13.2 from the package version in case of channel ga
+              if [[ "${{ inputs.CHANNEL }}" == "ga" ]]; then
```
Done

maork-elementor commented on .github/workflows/update-readme-txt/action.yml:
```diff
@@ -0,0 +1,36 @@
+              # Get the latest ga tag e.g. 3.13.2 from the package version in case of channel ga
+              if [[ "${{ inputs.CHANNEL }}" == "ga" ]]; then
+                latest_tag=${{ inputs.PACKAGE_VERSION }}
+              fi
```
Done

maork-elementor commented on .github/workflows/update-readme-txt/action.yml:
```diff
@@ -0,0 +1,36 @@
+              sed -i -E "s/Stable tag: .*/Stable tag: ${latest_tag}/g" $GITHUB_WORKSPACE/elementor/readme.txt
+
+              echo "readme.txt updated with Stable tag: ${latest_tag}"
+
+
+              
+              
+
```
Done

maork-elementor committed:
ci: Update readme.txt Stable tag with more details.

- Update script to update Stable tag in readme.txt
- Improve error messages for invalid channels or empty latest tag
- Made changes to [.github/workflows/update-readme-txt/action.yml]

maork-elementor committed:
Requested changes

maork-elementor committed:
update

maork-elementor committed:
Update GitHub action for stable tag in readme.txt

maork-elementor committed:
Refactor GitHub Actions workflow file formatting

yotamselementor commented on .github/workflows/update-readme-txt/action.yml:
```diff
@@ -0,0 +1,52 @@
+                # tail -n 1: This command selects the last (i.e., the highest) version number among the matched tags.
+                # sed 's/^refs\/tags\/v//': This command removes the leading "refs/tags/v" prefix from the selected tag name.
+
+                latest_tag=$(git ls-remote --tags | grep -v "\-rc" | grep -v "\-cloud" | grep -v "\-dev" | grep -v "\-beta" | awk '{print $2}' | grep -E '[0-9]+\.[0-9]+\.[0-9]+$' | sed 's/^refs\/tags\/v//' | sed 's/^refs\/tags\///' | sort -V | tail -n 1)
```
This fetches the same results for me, but I think its a bit simpler
```suggestion
                latest_tag=$(git ls-remote --tags | awk '{print $2}' | sed 's/^refs\/tags\/v//' | sed 's/^refs\/tags\///' | sort -V | grep -E '^[0-9]+\.[0-9]+\.[0-9]+$' | sort -V | tail -n 1)
```

maork-elementor committed:
Update .github/workflows/update-readme-txt/action.yml

Co-authored-by: yotamselementor <83912471+yotamselementor@users.noreply.github.com>

maork-elementor committed:
Merge branch 'main' into ED-10526
